File: src/shared/types/project.ts
import type { TaskPriority } from './taskPriority'
import type { TaskStatus } from './task'

export type Project = {
	id: string
	spaceId: string
	name: string
	description: string | null
	status: TaskStatus
	priority: TaskPriority
	plannedAt: string | null
	dueAt: string | null
	remindAt: string | null
	statusChangedAt: string
	position: number
	completedAt: string | null
	archivedAt: string | null
	deletedAt: string | null
	createdAt: string
	updatedAt: string
}

export type ProjectOverviewItem = Project & {
	spaceName: string
	taskCount: number
	activeTaskCount: number
}

export type ProjectSidebarItem = {
	id: string
	spaceId: string
	name: string
	position: number
	taskCount: number
	activeTaskCount: number
	completedAt: string | null
	updatedAt: string
}

File: src-tauri/crates/storage/src/entities/project.rs
//! Project entity。

use sea_orm::entity::prelude::*;

use super::common::WorkStatus;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
#[sea_orm(table_name = "projects")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: String,
    pub space_id: String,
    pub name: String,
    pub description: Option<String>,
    pub status: WorkStatus,
    pub priority: i32,
    pub planned_at: Option<String>,
    pub due_at: Option<String>,
    pub remind_at: Option<String>,
    pub status_changed_at: String,
    pub completed_at: Option<String>,
    pub position: i64,
    pub generation: i64,
    pub archived_at: Option<String>,
    pub deleted_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
